agent/lm_agent/parsing/lmx.py

```diff
@@ -0,0 +1,117 @@
+"""
+Parser for LM-X
+"""
+import re
+
+HOSTWORD = r"[a-zA-Z0-9-]+"
+HOSTNAME = rf"{HOSTWORD}(\.{HOSTWORD})*"
+INT = r"\d+"
+VERSION = rf"{INT}\.{INT}"
+
+FEATURE_LINE = rf"^Feature: (?P<feature>\w+) Version: (?P<version>{VERSION}) Vendor: (?P<vendor>\S+)$"
+IN_USE_LINE = rf"^(?P<in_use>{INT}) of (?P<total>{INT}) license\(s\) used(:)?$"
+USAGE_LINE = (
+    rf"^(?P<in_use>{INT}) license\(s\) "
+    rf"used by (?P<user>\S+)@(?P<lead_host>{HOSTNAME}) "
+    rf"\[{INT}\.{INT}\.{INT}\.{INT}\]"
+)
+RX_FEATURE = re.compile(FEATURE_LINE)
+RX_IN_USE = re.compile(IN_USE_LINE)
+RX_USAGE = re.compile(USAGE_LINE)
+
+
+def parse_feature_line(line: str):
+    """
+    Parse the feature line in the LM-X output.
+    Data we need:
+    - ``feature``: license name
+    """
+    parsed_feature = RX_FEATURE.match(line)
+    if parsed_feature is None:
+        return None
+    feature_data = parsed_feature.groupdict()
+    feature_data["feature"] = feature_data["feature"]
+
+    return feature_data["feature"]
```

Comment:
This is confusing. What's going on here?

Reply:
Oh, I forgot to remove this line.
I was using `.lower()` in the feature name before I realized that the LM-X licenses case non-consistent casing.
Thanks for pointing this!